### PR TITLE
Re-enable webkit tests

### PR DESCRIPTION
**Pull request by swissspidy**

### CODE REVIEW

**Comment by swissspidy** [comment]
I think the remaining ones are blocked by #611

**Comment by kleisauke** [comment]
I haven't been able to reproduce that error on non-x86 based architectures yet. As an experiment, we could try running the E2E WebKit tests on a [`macos-14` runner](https://github.com/actions/runner-images#available-images) to verify if this issue persists.

FWIW, hopefully they prioritize that WebKit bug report, as it can probably be used for fingerprinting (e.g. to distinguish Silicon-based Macs from Intel-based ones).

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -16,33 +16,37 @@ permissions: {}
...
+                      shard: '2/2'
+                      project: 'chromium'
                       experimental: true
 
         steps:
             - name: Checkout
               uses: actions/checkout@v4
 
+            - name: Install Docker (missing on macOS)
+              if: ${{ matrix.os == 'macos-latest' }}
```
Comment (kleisauke) [inline .github/workflows/e2e-tests.yml:45]:
```suggestion
              if: runner.os == 'macOS'
```

Comment (swissspidy) [reply .github/workflows/e2e-tests.yml:45]: Lol, this should have been `matrix.project == 'webkit'` of course 😅 🤦   But actually comparing the runner is even better.

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -16,33 +16,40 @@ permissions: {}
...
                       experimental: true
 
         steps:
             - name: Checkout
               uses: actions/checkout@v4
 
+            - name: Install Docker (missing on macOS)
+              if: runner.os == 'macOS'
+              run: |
+                brew install docker colima
```
Comment (kleisauke) [inline .github/workflows/e2e-tests.yml:47]: It looks like we may also need to install `docker-compose`.
```suggestion
                brew install docker docker-compose colima
```

Comment (swissspidy) [reply .github/workflows/e2e-tests.yml]: Just came to the same conclusion :) Let's see how it goes 🤞

File: .github/workflows/e2e-tests.yml [hunk 3]
```diff
@@ -16,33 +16,40 @@ permissions: {}
...
                       experimental: true
 
         steps:
             - name: Checkout
               uses: actions/checkout@v4
 
+            - name: Install Docker (missing on macOS)
+              if: runner.os == 'macOS'
+              run: |
+                brew install docker docker-compose colima
```
Comment (kleisauke) [inline .github/workflows/e2e-tests.yml:47]: It looks like there's still a caveat with `docker-compose` from Homebrew:
https://github.com/Homebrew/homebrew-core/blob/f0fe5a40bf3544c758cab6a8824cccc05c599dd3/Formula/d/docker-compose.rb#L39-L46

According to https://github.com/abiosoft/colima/discussions/874#discussioncomment-9582491 the old method of installing this plugin still works.
```suggestion
                brew install docker docker-compose colima
                ln -sfn $(brew --prefix docker-compose)/bin/docker-compose ~/.docker/cli-plugins/docker-compose
```

Which might be a bit easier.

Comment (kleisauke) [reply .github/workflows/e2e-tests.yml:47]: Ah, sorry, the initial suggestion was without the `mkdir` invocation, which seems to be necessary.

File: .github/workflows/e2e-tests.yml [hunk 4]
```diff
@@ -16,33 +16,44 @@ permissions: {}
...
+
+                # See https://github.com/abiosoft/colima/discussions/874
+                mkdir -p ~/.docker/cli-plugins
+                ln -sfn $(brew --prefix docker-compose)/bin/docker-compose ~/.docker/cli-plugins/docker-compose
+
+                # See https://github.com/abiosoft/colima/issues/970#issuecomment-2224011277
+                sudo curl -L -o /opt/homebrew/Cellar/lima/0.22.0/bin/limactl https://github.com/mikekazakov/lima-nohvf/raw/master/limactl && sudo chmod +x /opt/homebrew/Cellar/lima/0.22.0/bin/limactl
+                colima start --network-address --arch arm64 --vm-type=qemu
```
Comment (kleisauke) [inline .github/workflows/e2e-tests.yml:55]: Looking at the recent commits of Colima (e.g commit https://github.com/abiosoft/colima/commit/aa4d3e7780496e069fea43facff0f55b4bc67f34), I think one could just use `--vm-type=vz` out-of-the-box.
```suggestion
                colima start --network-address --arch arm64 --vm-type=vz --cpu 3 --memory 7
```
(I also added `--cpu 3 --memory 7` to match [the hardware used by M1 macOS runner](https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/))

If that doesn't work, it should probably run under Rosetta 2 emulation.
```suggestion
                colima start --network-address --arch arm64 --vm-type=vz --vz-rosetta --cpu 3 --memory 7
```
https://github.com/abiosoft/colima/blob/73732ae8e44148b6191406f61044eff8195ae29b/README.md?plain=1#L152-L156

Comment (swissspidy) [reply .github/workflows/e2e-tests.yml]: Thank you so much for helping me debug this!

If we can't get it to work, I'm exploring a Docker-less alternative in #614 to make things a bit easier.

Comment (swissspidy) [reply .github/workflows/e2e-tests.yml]: Neither seems to work 😞 🤔

Comment (kleisauke) [reply .github/workflows/e2e-tests.yml]: Hmm, it looks like `--vm-type=vz` also uses nested virtualization. :( 
```
{"level":"fatal","msg":"Error Domain=VZErrorDomain Code=2 Description=\"Invalid virtual machine configuration. Virtualization is not available on this hardware.\" UserInfo={\n    NSLocalizedFailure = \"Invalid virtual machine configuration.\";\n    NSLocalizedFailureReason = \"Virtualization is not available on this hardware.\";\n}","time":"2024-08-14T16:08:38Z"}
````
https://github.com/kleisauke/media-experiments/actions/runs/10391084038/job/28773154085#step:3:214

> I'm exploring a Docker-less alternative in #614 to make things a bit easier.

Ah, that would be neater. FWIW, Podman isn't supported on the `macos-14` runner either:
https://github.com/actions/runner-images/discussions/9304

**Comment by kleisauke** [comment]
After some thought, perhaps it might be just easier to pass the `JSC_webAssemblyLLIntTiersUpToBBQ=0` env variable when running the WebKit tests on Ubuntu? I had a quick go with commit https://github.com/kleisauke/media-experiments/commit/f428a27a433a29d72743d8cfe5242fbbc54fc091, which seems to work at first glance.

**Pull request merged**